packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.tsx

```diff
@@ -184,29 +163,30 @@ export const BaseTextEditor = forwardRef<EditorView, TextEditorProps>(
         newView?.destroy();
         setView(null);
       };
-    }, [rootRef, model, readonly, editorMode, themeMode]);
+    }, [rootRef, model, readonly, themeMode]);
 
-    // Handles tab/focus.
-    // Pressing Escape focuses the outer div (to support tab navigation); pressing Enter refocuses the editor.
-    // TODO(burdon): Convert to keymap?
-    const handleKeyUp = useCallback(
-      (event: KeyboardEvent) => {
+    // Pressing Escape focuses the outer div (to support tab navigation). Pressing Enter should refocus the editor.
+    // In vim-mode, press a key modifier to focus the outer div.
+    const handleKeyUp = useCallback<KeyboardEventHandler<HTMLDivElement>>(
+      (event) => {
         const { key, altKey, shiftKey, metaKey, ctrlKey } = event;
         switch (key) {
           case 'Enter': {
             view?.focus();
             break;
           }
-
           case 'Escape': {
-            if (editorMode === 'vim' && (altKey || shiftKey || metaKey || ctrlKey)) {
+            if (altKey || shiftKey || metaKey || ctrlKey) {
+              rootRef.current?.focus();
+              break;
+            } else {
               view?.focus();
             }
             break;
           }
```

Comment:
I only added this for the vim extension so it should probably be factored out as well. Maybe pairing the vim extension w/ a keymap extension that has this functionality?

Reply:
the CM keymap maps onto keypress not up/down so it can't handle ESCAPE, nor does it have access to the outer div. So my approach here was to treat the outer div as a container that takes the tabster attributes and handles external navigation.

Reply:
i think i've properly factored this out now.